src/components/Dashboard/scenes/ManageMentorApplications/ManageMentorApplications.tsx

```diff
@@ -1,11 +1,223 @@
-import React from 'react';
+import React, { useState, useEffect } from 'react';
+import { type Mentor } from '../../../../types';
+import { API_URL } from '../../../../constants';
+import axios from 'axios';
+
+export const ManageMentorApplications: React.FC = () => {
+  const [mentors, setMentors] = useState<Mentor[]>([]);
+  const [filter, setFilter] = useState('');
+  const [loading, setLoading] = useState(false);
+  const [searchTerm, setSearchTerm] = useState('');
+  const [categories, setCategories] = useState<string[]>([]);
+  const [categoryFilter, setCategoryFilter] = useState<string>('');
+
+  const renderFilters = () => {
+    const filters = [
+      { label: 'All', status: '' },
+      { label: 'Approved', status: 'approved' },
+      { label: 'Pending', status: 'pending' },
+      { label: 'Rejected', status: 'rejected' },
+    ];
+
+    return (
+      <div className="flex mb-4">
+        {filters.map(({ label, status }) => {
+          const count = mentors.filter((mentor) =>
+            status.length > 0 ? mentor.state === status : true
+          ).length;
+          return (
+            <button
+              key={label}
+              className={`px-4 py-2 font-medium text-sm ${
+                filter === status
+                  ? 'bg-blue-500 text-white'
+                  : 'bg-gray-200 text-gray-700 hover:bg-gray-300'
+              } ${
+                label === 'All'
+                  ? 'rounded-l-md'
+                  : label === 'Rejected'
+                  ? 'rounded-r-md'
+                  : 'rounded-none'
+              }`}
+              onClick={() => {
+                setFilter(status);
+              }}
+            >
+              {label} ({count})
+            </button>
+          );
+        })}
+      </div>
+    );
+  };
+
+  const handleStatusUpdate = async (mentorId: string, newStatus: string) => {
+    try {
+      const response = await axios.put(
+        `${API_URL}/admin/mentors/${mentorId}/status`,
+        { status: newStatus },
+        { withCredentials: true }
+      );
+      if (response.status === 200) {
+        void fetchMentors();
+      } else {
+        console.error('Failed to update mentor status');
+      }
+    } catch (error) {
+      console.error('Error updating mentor status:', error);
+    }
```

Comment:
use react query here. https://tanstack.com/query/latest/docs/framework/react/guides/queries

Reply:
Thanks! First-time case across this. seems super awesome.